pkg/collectors/pool.go

```diff
@@ -127,6 +136,131 @@ func (f *PerfCollector) initPoolDescs() {
+			if disk[ControllerNameKey].(string) != "" {
+				internalPool = false
+			} else {
+				internalPool = true
+			}
+		}
+	}
+	return internalPool, nil
+}
+
+func isCompressionEnabled(poolName string, disksList rest.MDisksList, fsRestClient *rest.FSRestClient) (bool, error) {
+	var compressionEnabled bool
+	disksInPool := mapMDisksToPool(poolName, disksList)
+
+	for _, diskID := range disksInPool {
+		MDiskInfo, err := fsRestClient.LsSingleMDisk(diskID)
+		if err != nil {
+			log.Errorf("get single mdisk info error: %v", err)
+			return compressionEnabled, err
+		}
+		if MDiskInfo[MdiskEffectiveUsedCapacity].(string) == "" {
+			compressionEnabled = false
+		} else {
+			compressionEnabled = true
+		}
+	}
+	return compressionEnabled, nil
+}
+
+func isPoolArrayMode(poolName string, disksList rest.MDisksList) (bool, error) {
+	var arrayMode bool
+	for _, disk := range disksList {
+		if poolName == disk[MdiskGroupNameKey].(string) {
+			if disk[DiskModekey].(string) != "array" {
+				arrayMode = false
+			} else {
+				arrayMode = true
+			}
+		}
+	}
+	return arrayMode, nil
+}
+
+func (f *PerfCollector) CalcReducedReclaimableCapacityForPool(pool Pool, fsRestClient *rest.FSRestClient, mDisksList rest.MDisksList) (float64, error) {
+	var totalDisksCapacities float64
+	var midSum float64
+	disksInPool := mapMDisksToPool(pool[MdiskNameKey].(string), mDisksList)
+	reclaimable, err := strconv.ParseFloat(pool[ReclaimableKey].(string), 64)
+	if err != nil {
+		log.Errorf("get pool reclaimable capacity failed: %s", err)
+		return -1, err
+	}
+
+	for _, diskID := range disksInPool {
+		PC, EU, physicalFree, err := calcCapacityForSingleDisk(fsRestClient, diskID)
+		if err != nil {
+			log.Errorf("get single disk capacity failed: %s", err)
+			return -1, err
+		}
+		PU := math.Max(0, PC-physicalFree)
+		diskRatio := PC * PU / EU
+
+		totalDisksCapacities += PC
+		midSum += diskRatio
+
+		log.Infof("Calculating reduced reclaimable capacity for Disk ID: %d related to pool %v, "+
+			"PhysicalCapacity PC: %f, MdiskEffectiveUsedCapacity EU: %f, PU: %f, diskRatio: %f, totalDisksCapacities: %f, midSum: %f",
+			diskID, pool[MdiskNameKey].(string), PC, EU, PU, diskRatio, totalDisksCapacities, midSum)
+	}
+
+	if totalDisksCapacities == 0 || midSum == 0 {
+		return 0, nil
+	} else {
+		return reclaimable / totalDisksCapacities * midSum, nil
+	}
+}
+
+func mapMDisksToPool(poolName string, disksList rest.MDisksList) []int {
+	var disksInPool []int
+	for _, disk := range disksList {
+		if poolName == disk[MdiskGroupNameKey].(string) {
+			diskId, _ := strconv.Atoi(disk[MdiskIdKey].(string))
+			disksInPool = append(disksInPool, diskId)
+		} else {
+			continue
+		}
+	}
+	return disksInPool
+}
+
+func calcCapacityForSingleDisk(fsRestClient *rest.FSRestClient, diskID int) (float64, float64, float64, error) {
+	MDisksInfo, err := fsRestClient.LsSingleMDisk(diskID)
+	if err != nil {
+		log.Errorf("get disk list error: %v", err)
+		return -1, -1, -1, err
```

Comment:
Maybe we should use const for that -1 value, to indicate an invalid capacity value (so in the future it will be easier to change the value or understand its purpose. What do you think?

Reply:
fixed.